### PR TITLE
UI improvements

### PR DESCRIPTION
**Pull request by simmac**
There is still a lot to be done, but I tried to solve the worst UI problems.

### CODE REVIEW

**Comment by dbrgn** [comment]
Looks much better than before!

Another suggestion: Would it be possible to make the bubbles only as high as necessary? Right now a single message is stretched vertically to fill the entire chat area, with a lot of whitespace in it.

**Comment by blizzard4591** [comment]
> Would it be possible to make the bubbles only as high as necessary?

Sadly, that is not as easy as it sounds, because of the way Qt calculates the height of contents in the scrollable area.
I hope that with a simple HTML5 UI this will be a problem of the past.

File: src/main.ui [hunk 1]
```diff
@@ -6,15 +6,64 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>600</width>
-    <height>454</height>
+    <width>561</width>
+    <height>400</height>
```
Comment (blizzard4591) [inline src/main.ui]: Is this change intentional?

**Comment by blizzard4591** [comment]
Wow, this looks much better. 
A few small issues:

- Messages from ownself (or at least their time/from line) are clipped by the rounded corner.
- If the windows is not wide enough, the time/from line wraps, but only one line is ever displayed.

**Comment by simmac** [comment]
@blizzard4591 
> Is this change intentional?
Not really, I reverted it. I tried to make the window smaller, but for some reason I can't resize it to less than 1058px on OS X. Do you have any idea?

> Messages from ownself (or at least their time/from line) are clipped by the rounded corner.
Could you show me a screenshot? On my OS X machine, I can't reproduce this.

> If the windows is not wide enough, the time/from line wraps, but only one line is ever displayed.
Should be fixed now (?) but I can't test it because, as I said, I can't make the window small enough.

**Comment by blizzard4591** [comment]
openmittsubugpullreq3

Ahh, I stand corrected - the issue is not the rounded corner, but instead it looks like Qt is not calculating the width correctly for the italic font?!

**Comment by simmac** [comment]
Oh, I see now.
Maybe this is helpful: https://stackoverflow.com/questions/23737591/qpainter-drawtext-italic-writes-outside-of-the-rectangle
If you have an idea, let me know, otherwise I'll try to somehow work around this soon.

**Comment by dbrgn** [comment]
By the way, why aren't you using QML? (This is a real question, not a critique, I never worked with "classic" QT or QML before.)

**Comment by simmac** [comment]
Hmm, thanks to @KL7000F I just realized text wrapping doesn't really work as intended. On my OS X machine, long text just gets cut at line end and not wrapped around at all. 
On Ubuntu , the text wraps at line end, but it somehow the text gets cut off after two lines but expands again after a new message is shown under the current message.

**Comment by KL7000F** [comment]
Here is a Screenshot from Ubuntu that shows the error:
openmittsu
Thanks @simmac

**Comment by blizzard4591** [comment]
Take a look at the changes in https://github.com/blizzard4591/openMittsu/tree/uiEditBySimmacPullReq3

This should fix the font issues. Still investigating on the Line Break/Resize stuff.

**Comment by blizzard4591** [comment]
Meh, just realized that this probably does not fix every issue. If the line gets wrapped, all wrapped lines will still be cut off.
Margin and Padding do not work here, since they are "outside" the text-area of the Label. 
I will probably file a bug report for this, the label should solve this problem, not the developer using it...

**Comment by KL7000F** [comment]
The current changes from simmac looks fine under ubuntu. The italic text now good at the end.

**Comment by blizzard4591** [comment]
At the end - yes. But not if a line break occures:
openmittsubugpullreq3_2

**Comment by KL7000F** [comment]
Damn, your right! But, is it not a better idea to shorten / cut the text? In the app we have only the time from receive and the agreed / disagreed. I would put the name and ID on the top and shorten the time / date (with year?) / status to a minimum. Then we have no more problems.

**Comment by blizzard4591** [comment]
I tried building a minimal example for a bug report and can not get it to fail as nicely as on the screenshots. There seem to be special cicumstances here. Ideas?

**Comment by KL7000F** [comment]
What's the big problem?

**Comment by blizzard4591** [comment]
Now tracking externally under https://bugreports.qt.io/browse/QTBUG-53502

The minimal example required the text-alignment attribute "AlignRight" to trigger the bug.

**Comment by blizzard4591** [comment]
Ok. Since your UI changes are way to good to keep them from the users, I will merge this and fix the italic font issue later on :)
